server/src/main/java/org/opensearch/index/store/lockmanager/RemoteStoreMetadataLockManager.java

```diff
@@ -70,6 +72,15 @@ public void release(LockInfo lockInfo) throws IOException {
         }
     }
 
+    public String fetchLock(String filenamePrefix, String acquirerId) throws IOException {
+        Collection<String> lockFiles = lockDirectory.listFilesByPrefix(filenamePrefix);
+        List<String> lockFilesForAcquirer = lockFiles.stream()
```

Comment:
nit: there should be only one lock for an acquirerId, we can pull the first available one. and assert that there are only 1 lock for a given acquirer id.